document sandfly security connector

seanstory opened the pull request:
Adds documentation for the connector added in https://github.com/elastic/connectors/pull/3522

seanstory committed:
Stubbed out sandfly connector docs

seanstory committed:
Caught some minor changes in review

seanstory committed:
Merge branch 'main' into seanstory/document-sandfly-connector

leemthompo commented on docs/reference/search-connectors/es-connectors-sandfly.md:
```diff
@@ -0,0 +1,315 @@
+
+This connector is available as a self-managed connector.
+
+This self-managed connector is compatible with Elastic versions **8.19.0+**.
```
```suggestion
This self-managed connector is compatible with Elastic versions **8.19.0/9.1.0+**.
```

leemthompo reviewed (commented):
LGTM content wise, but we need to add this page to the TOC for it to get built:

https://github.com/elastic/elasticsearch/blob/main/docs/reference/search-connectors/toc.yml

These are the 9.x docs, do we also need an equivalent in 8.x asciidoc for 8.19? 

Will let devs approve on technical details :)

seanstory committed:
Add to TOC

seanstory committed:
Specify 9.1.0+ support

seanstory commented:
> we need to add this page to the TOC for it to get built

fixed! Thanks!

> These are the 9.x docs, do we also need an equivalent in 8.x asciidoc for 8.19?

I vote "no" actually. The push for the new connectors will be for the new solution type, not to offer these connectors for older versions. So I've adjusted the doc to say it's for 9.1.0+

leemthompo reviewed (approved):
LGTM from docs structure perspective 👍
